### PR TITLE
add location categories

### PR DESCRIPTION
**Pull request by kcchik**
```
python manage.py migrate
```

add categories
```
~ ➜ curl \
-X POST \
-H 'Content-Type: application/json' \
-H 'Accept: application/json' \
-d '{"categories":["1","3"]}' \
localhost:8000/api/locations/1/categories/
{"id":1,"name":"mcdonalds","longitude":0.0,"latitude":0.0,"photos":[4,5],"tags":[],"categories":[1,2,3],"created":"2022-07-25T17:38:44.747263Z"}%
```

delete categories
```
~ ➜ curl \
-X DELETE \
-H 'Content-Type: application/json' \
-H 'Accept: application/json' \
-d '{"categories":["1","3"]}' \
localhost:8000/api/locations/1/categories/
```

### CODE REVIEW

File: api/views/location.py [hunk 1]
```diff
@@ -45,42 +48,68 @@ def get_queryset(self):
...
+
+class LocationCategories(generics.GenericAPIView):
+    serializer_class = LocationSerializer
+    queryset = Location.objects.all()
+
+    def post(self, request, *args, **kwargs):
+        instance = self.get_object()
+        category_ids = request.data.get('categories')
+        if category_ids:
+            categories = Category.objects.filter(id__in=category_ids)
```
Comment (Raymond-Xia) [inline api/views/location.py:61]: Should add some error handling, if you try to add a category that doesn't already exist, the request returns 200 but categories in the location doesn't get updated which is a bit confusing.

File: api/serializers.py [hunk 2]
```diff
@@ -10,7 +10,7 @@ class CategorySerializer(serializers.ModelSerializer):
 
     class Meta:
         model = Category
-        fields = ['id', 'name', 'tags', 'created']
+        fields = ['id', 'name', 'tags', 'locations', 'created']
```
Comment (Raymond-Xia) [inline api/serializers.py:13]: Seems to be causing issues, I thought we weren't adding locations to the categories?
image

**Review by Raymond-Xia — CHANGES_REQUESTED** [review]

File: api/views/location.py (hunk 1 above)
Comment (kcchik) [reply api/views/location.py:61]: what if we add a list of categories and only one doesnt exist

Comment (Raymond-Xia) [reply api/views/location.py:61]: hmm true, might be annoying but I'd say reject the whole request and say that the one category doesn't exist

**Review by Raymond-Xia — APPROVED** [review]

**Pull request merged**